From e0a70c26b2953d8f54b53657fdb5a350a933120e Mon Sep 17 00:00:00 2001
From: isyippee <yippee_liu@163.com>
Date: Wed, 11 Nov 2015 10:53:44 +0800
Subject: [PATCH 05/14] Determine the installation of httpd

---
 ceph_deploy/eayunrgw.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ceph_deploy/eayunrgw.py b/ceph_deploy/eayunrgw.py
index cea8ddd..ad8b9c0 100644
--- a/ceph_deploy/eayunrgw.py
+++ b/ceph_deploy/eayunrgw.py
@@ -167,7 +167,9 @@ def config_http(distro, conn, gw_name):
     remoto.process.run(conn, ['mkdir', '-p', '/var/lib/ceph/radosgw/ceph-radosgw.%s' % gw_name, ],
                        timeout=7)
     # install httpd
-    remoto.process.run(conn, ['yum', 'install', '-y', 'httpd', ], timeout=0)
+    stdout, stderr, returncode = remoto.process.check(conn, ['rpm','-qa','httpd'], timeout=7)
+    if not stdout:
+        remoto.process.run(conn, ['yum', 'install', '-y', 'httpd', ], timeout=0)
     remoto.process.run(conn, ['chown', 'apache:apache', '/var/run/ceph', ], timeout=0)
     distro.conn.remote_module.touch_file('/var/log/radosgw/client.radosgw.gateway.log')
     remoto.process.run(conn, ['chown', 'apache:apache', '/var/log/radosgw/client.radosgw.gateway.log', ],
-- 
2.8.0.rc3

