From 4db380091f9e2e3bd2b8deaeab006c25066a1e98 Mon Sep 17 00:00:00 2001
From: isyippee <yippee_liu@163.com>
Date: Wed, 11 Nov 2015 11:15:48 +0800
Subject: [PATCH 06/14] a little problem about syntax,var

---
 ceph_deploy/eayunrgw.py | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/ceph_deploy/eayunrgw.py b/ceph_deploy/eayunrgw.py
index ad8b9c0..526e849 100644
--- a/ceph_deploy/eayunrgw.py
+++ b/ceph_deploy/eayunrgw.py
@@ -109,10 +109,12 @@ def new_rgw_keyring(args, gw_name, conn):
     keyring_path = '{name}.client.radosgw.keyring'.format(
         name=args.cluster,
     )
-    try:
-        rgw_keyring = files.read_file(keyring_path)
-    except Exception:
-        rgw_keyring = ''
+    rgw_keyring = ''
+    if os.path.exists(keyring_path):
+        try:
+            rgw_keyring = files.read_file(keyring_path)
+        except Exception:
+            raise 
 
     rgw_keyring += '''[client.radosgw.%s]
     key = %s
@@ -151,12 +153,12 @@ def new_rgw_keyring(args, gw_name, conn):
         [
             'ceph',
             '-k',
-            '/etc/ceph/ceph.client.admin.keyring',
+            '/etc/ceph/%s.client.admin.keyring' % args.cluster,
             'auth',
             'add',
             'client.radosgw.%s' % gw_name,
             '-i',
-            '/etc/ceph/ceph.client.radosgw.keyring',
+            '/etc/ceph/%s.client.radosgw.keyring' % args.cluster,
         ],
         timeout=7
     )
-- 
2.8.0.rc3

