From 92b758bee417621fb6f6f90e45d272a779d2d6fc Mon Sep 17 00:00:00 2001
From: isyippee <yippee_liu@163.com>
Date: Wed, 16 Dec 2015 10:15:48 +0800
Subject: [PATCH 08/14] Add pools

Signed-off-by: isyippee <yippee_liu@163.com>
---
 ceph_deploy/eayunrgw.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/ceph_deploy/eayunrgw.py b/ceph_deploy/eayunrgw.py
index ac3d7f0..b51e13a 100644
--- a/ceph_deploy/eayunrgw.py
+++ b/ceph_deploy/eayunrgw.py
@@ -114,7 +114,7 @@ def new_rgw_keyring(args, gw_name, conn):
         try:
             rgw_keyring = files.read_file(keyring_path)
         except Exception:
-            raise 
+            raise
 
     rgw_keyring += '''[client.radosgw.%s]
     key = %s
@@ -244,7 +244,10 @@ def eayunrgw_create(args):
     new_rgw_keyring(args, gw_name, conn)
 
     # Create pools.
-    suffix = ['.rgw.root', '.rgw.control', '.rgw.gc', '.log', '.users', '.users.email', '.users.uid']
+    suffix = ['rgw', '.rgw.root', '.rgw.control', '.rgw.gc',
+              'rgw.buckets', 'rgw.buckets.index', 'rgw.buckets.extra',
+              '.log', 'intent-log', 'usage', '.users',
+              '.users.email', '.users.uid']
     pools = ['.%s%s' % (zone_name, suf) for suf in suffix]
     for pool in pools:
         remoto.process.run(conn,
-- 
2.8.0.rc3

