From 2dca7eabf0aee785b05dc880e78aa9c31bdd552f Mon Sep 17 00:00:00 2001
From: isyippee <yippee_liu@163.com>
Date: Wed, 16 Dec 2015 10:23:16 +0800
Subject: [PATCH 10/14] Modify rgw start method and enable httpd

Signed-off-by: isyippee <yippee_liu@163.com>
---
 ceph_deploy/eayunrgw.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ceph_deploy/eayunrgw.py b/ceph_deploy/eayunrgw.py
index b8cff40..66fd741 100644
--- a/ceph_deploy/eayunrgw.py
+++ b/ceph_deploy/eayunrgw.py
@@ -85,9 +85,9 @@ def get_hosts(cfg, zone_name):
 def restart_serv(conn, gw_name):
     remoto.process.run(conn, ['/etc/init.d/ceph', 'restart', 'mon', ], timeout=7)
     remoto.process.run(conn,
-                       ['radosgw', '-c', '/etc/ceph/ceph.conf', '-n', 'client.radosgw.%s' % gw_name,
-                        '--debug-rgw', '-d'],
+                       ['/etc/init.d/ceph-radosgw', '-n', 'client.radosgw.%s' % gw_name],
                        timeout=7)
+    remoto.process.run(conn, ['systemctl', 'enable', 'httpd', ], timeout=7)
     remoto.process.run(conn, ['systemctl', 'restart', 'httpd', ], timeout=7)
 
 
-- 
2.8.0.rc3

