From 1344606b2fc82465b48917ccb8e9c27874c4d11b Mon Sep 17 00:00:00 2001
From: Zhao Chao <zhaochao1984@gmail.com>
Date: Thu, 31 Dec 2015 17:48:08 +0800
Subject: [PATCH 16/21] eayunrgw: change rgw backend listening port to 8080.

Signed-off-by: Zhao Chao <zhaochao1984@gmail.com>
---
 ceph_deploy/eayunrgw.py | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/ceph_deploy/eayunrgw.py b/ceph_deploy/eayunrgw.py
index 63451aa..f3121f5 100644
--- a/ceph_deploy/eayunrgw.py
+++ b/ceph_deploy/eayunrgw.py
@@ -14,7 +14,7 @@ from ceph_deploy.util import files
 
 LOG = logging.getLogger(__name__)
 
-RGW_CONTECT = '''<VirtualHost *:80>
+RGW_CONTECT = '''<VirtualHost *:8080>
     ServerName localhost
     DocumentRoot /var/www/html
     ErrorLog /var/log/httpd/rgw_error.log
@@ -32,12 +32,12 @@ REGION_CONTECT = '''{ "name": "%s",
     "api_name": "%s",
     "is_master": "true",
     "endpoints": [
-          "http:\/\/%s:80\/"],
+          "http:\/\/%s:8080\/"],
     "master_zone": "%s",
     "zones": [
         { "name": "%s",
             "endpoints": [
-                "http:\/\/%s:80\/"],
+                "http:\/\/%s:8080\/"],
             "log_meta": "true",
             "log_data": "true"}],
     "placement_targets": [
@@ -237,7 +237,9 @@ def config_http(distro, conn, gw_name):
     remoto.process.run(conn, ['chown', 'apache:apache', '/var/log/radosgw/client.radosgw.gateway.log', ],
                        timeout=0)
     httpd_path = '/etc/httpd/conf/httpd.conf'
-    httpd_context = '%s\n%s' % (distro.conn.remote_module.get_file(httpd_path), FACTCGI_CONTECT)
+    httpd_conf = distro.conn.remote_module.get_file(httpd_path)
+    httpd_conf = httpd_conf.replace('\nListen 80\n', '\nListen 8080\n')
+    httpd_context = '%s\n%s' % (httpd_conf, FACTCGI_CONTECT)
     distro.conn.remote_module.write_file(httpd_path, httpd_context)
     gw_context = RGW_CONTECT % gw_name
     gw_path = '/etc/httpd/conf.d/rgw.conf'
-- 
2.8.0.rc3

